apps/src/templates/sectionsRefresh/CurriculumQuickAssign.jsx

```diff
@@ -143,44 +148,38 @@ export default function CurriculumQuickAssign({
         hasTextToSpeech: targetUnit?.text_to_speech_enabled,
       };
 
-      updateSection('course', updateSectionData);
+      updateCourse(updateSectionData);
     },
-    [updateSection, sectionCourse]
+    [updateCourse, sectionCourse]
   );
 
   /*
-  When toggling 'decide later', clear out marketing audience or assign one to make
-  the table appear again automatically.
-  Additionally, erase any previously selected course assignment.
+    When toggling 'decide later', erase any selected course assignment.
+    Leave the marketing audience alone to prevent toggling of the table that
+    might be jarring to the user.
   */
   const toggleDecideLater = () => {
-    setDecideLater(!decideLater);
-    updateSection('course', {});
-    if (marketingAudience !== '') {
-      setMarketingAudience('');
-      setSelectedCourseOffering(null);
-    } else {
-      setMarketingAudience(MARKETING_AUDIENCE.ELEMENTARY);
+    // User clicked "Clear assigned curriculum"
```

Comment:
I'm not sure these comments make sense with the logic here. Would we ever set decideLater to false if the text was not 'Clear assigned curriculum'? I think what would make more sense is to keep a clear toggle (always setting to !decideLater without an if statement) or setting it false in one case, and true in the other.

Reply:
The text is determined by on line 193: if `selectedCourseOffering` is truthy the text is "Clear assigned curriculum", otherwise it's "Decide later". That's where these comments are coming from: this function is called when the user clicks the button, and the value of `selectedCourseOffering` corresponds to what the text on that button was when they clicked it.

Meanwhile the `decideLater` variable holds the state of the checkbox. What I'm trying to do here is make sure if they clicked "Clear assigned curriculum" the button changes to "Decide later" with an unselected (`false`) checkbox. If they clicked "Decide later" it's a true toggle, flipping the checkbox to the opposite value.

Reply:
Yes... I suppose we're overloading the functionality of this checkbox anyway. Works for me!